agent/unit-test/softwarecontaineragent_unittest.cpp

```diff
@@ -53,12 +53,33 @@ class SoftwareContainerAgentTest: public ::testing::Test
     }
 };
 
-TEST_F(SoftwareContainerAgentTest, CreateContainerWithNoConf) {
-    ContainerID id = sca->createContainer("iejr-", "");
-    // This is actually only true if no other containers have been created
-    // before this one. Might need to be fixed somehow.
-    ASSERT_TRUE(id == 0);
+TEST_F(SoftwareContainerAgentTest, CreatAndCheckContainer) {
+	SoftwareContainer *container;
+	ContainerID id = sca->createContainer("iejr-", "");
+    bool retval = sca->checkContainer(id, container);
+    ASSERT_TRUE(retval == true);
+}
+
+TEST_F(SoftwareContainerAgentTest, DeleteContainer) {
+	SoftwareContainer *container;
+	ContainerID id = sca->createContainer("iejr-", "");
+	sca->deleteContainer(id);
+	bool retval = sca->checkContainer(id, container);
+	ASSERT_TRUE(retval == false);
 }
+/*
+ * Stress test written for creation + deletion
+ * It takes too much time, thus commented out
+ *
+TEST_F(SoftwareContainerAgentTest, StressCreateDeleteContainer) {
+	ContainerID id = 0;
+	for (auto i = 0 ; i < UINT32_MAX; i++) {
```

Comment:
If we have stress tests that would require a lot of time or some special treatment I think we should put them in separate suites and make it possible to deselect them easily so the CI system essentially can choose to run them and developers can choose to skip. W.r.t. unit-tests or not, these types of tests will IMHO fall into the third, slightly undefined, category between proper unit-tests and the service-tests. In any case, are stress testing the creation of container related to container ID? This should perhaps be done with a different scope.